contracts/Proposal.sol

```diff
@@ -0,0 +1,120 @@
+pragma solidity ^0.5.8;
+
+import "./zeppelin/token/ERC20/IERC20.sol";
+import "./zeppelin/token/ERC20/SafeERC20.sol";
+import "./ownership/Ownable.sol";
+import "./Basket.sol";
+
+/**
+ * A Proposal represents a suggestion to change the backing for RSV.
+ *
+ * The lifecycle of a proposal:
+ * 1. Creation
+ * 2. Acceptance
+ * 3. Completion
+ *
+ * A time can be set during acceptance to determine when completion is eligible.
+ * A proposal can also be closed before it is Completed. 
+ *
+ * This contract is intended to be used in one of two possible ways. Either:
+ * - A target RSV basket is suggested, and quantities to be exchanged are  
+ *     deduced at the time of proposal execution.
+ * - A specific quantity of tokens to be exchanged is suggested, and the 
+ *     resultant RSV basket is determined at the time of proposal execution.
+ *
+ */
+contract Proposal is Ownable {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    uint256 public id;
+    uint256 public time;
+    address public proposer;
+    address[] public tokens;
+    uint256[] public quantitiesIn; // token quantities to be added to the Vault
+    uint256[] public quantitiesOut; // total quantities to be withdrawn from the Vault
+
+    enum Statuses { Created, Accepted, Closed, Completed }
+    Statuses status;
+
+    uint256[] basketBacking;
+    Basket public basket;
+
+
+    constructor(
+        uint256 _id,
+        address _proposer,
+        address[] memory _tokens,
+        uint256[] memory _quantitiesIn, // total quantities for the entire RSV supply, not per RSV
+        uint256[] memory _quantitiesOut, // total quantities for the entire RSV supply, not per RSV
+        Basket _basket
+    ) 
+        public 
+    {
+        require(_quantitiesIn.length == _quantitiesOut.length, "quantities mismatched");
+        (id, proposer, tokens, quantitiesIn, quantitiesOut, basket) = 
+            (_id, _proposer, _tokens, _quantitiesIn, _quantitiesOut, _basket);
```

Comment:
I would prefer writing these vertically. Easier to scan and see that they are matched up correctly.

Reply:
Done.